File: src/main/scala/kafkastreams/scalaexamples/HelloKafkaStreams.scala
package kafkastreams.scalaexamples

import org.apache.kafka.streams.Topology
import org.apache.kafka.streams.scala.ImplicitConversions._
import org.apache.kafka.streams.scala.Serdes.String
import org.apache.kafka.streams.scala.StreamsBuilder

object HelloKafkaStreams extends App {
  new HelloKafkaStreams().start("hello-kafka-streams")
}

class HelloKafkaStreams extends KafkaStreamsApp {
  def createTopology(builder: StreamsBuilder): Topology = {
    builder.stream[String, String]("names")
      .mapValues(name => s"Hello, $name!")
      .to("hello")

    builder.build()
  }
}

File: src/main/java/kafkastreams/javaexamples/HelloKafkaStreams.java
package kafkastreams.javaexamples;

import org.apache.kafka.common.serialization.Serde;
import org.apache.kafka.common.serialization.Serdes;
import org.apache.kafka.streams.StreamsBuilder;
import org.apache.kafka.streams.Topology;
import org.apache.kafka.streams.kstream.Consumed;
import org.apache.kafka.streams.kstream.Produced;

public class HelloKafkaStreams extends KafkaStreamsApp {

    public static void main(String[] args) {
        new HelloKafkaStreams().start("hello-world-app");
    }

    public Topology createTopology(StreamsBuilder builder) {
        Serde<String> strings = Serdes.String();

        builder.stream("names", Consumed.with(strings, strings))
                .mapValues(name -> "Hello, " + name + "!")
                .to("hello", Produced.with(strings, strings));

        return builder.build();
    }
}
